### PR TITLE
Automating Testing

### LINKED ISSUE #70: Automating Backend Testing
**Problem**: Too much pulling down branches locally to test :<
**Solution**: Adding a cool and hip GitHub Action as a CI/CD pipeline to automate testing

**Description**: Now that we set up the Jest framework on the backend, we can automate the testing process by setting up a CI/CD pipeline with GitHub Actions. The idea is that every time a new pull request is made/committed, GitHub will run the unit tests for reviewers. This way reviewers like our great leader Sophie won't have to pull down every branch to test locally. 
This would require some research into GitHub Actions and the configurations to set up the job along with some dummy prs to test it out as I would imagine.

**Specs:**
- [x] research GitHub Actions
- [x] Setting up the testing pipeline for Jest
- [x]  more research into Dockerizing database or the entire project 🤔 
- [x] testing it out on a few dummy pull requests

### PR DESCRIPTION
**Pull request by VKong6019**
~(how do i add a card)~ I figured it out :D

This `yml` file sets up a GitHub Action to run `npm test` every time a pull request is made/committed in our backend repo. I'm thinking of also setting up linter (eslint) in a separate pr.

These are the docs that I found helpful in setting this up, 
- https://docs.github.com/en/actions/learn-github-actions/introduction-to-github-actions
- https://docs.github.com/en/actions/guides/building-and-testing-nodejs
- https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request
- https://docs.github.com/en/actions/guides/creating-postgresql-service-containers#running-jobs-in-containers

This finally works oml.
A couple things when this is merged, the `ormconfig.json` is now a `ormconfig.ts` file, so that the `test.yml` file can use environment variables. Generally, this should make it easier for ormconfig changes and all secrets should be in a local `.env` file in the project root. 

I can also update the existing docs that reference `ormconfig.json` to be `ormconfig.ts` and also create a `.env` file instead.

### CODE REVIEW

**Comment by VKong6019** [comment]
one thing i just realized that may run into errors is the postgres db part that i may have to dig deeper into

maybe this is when we'll have docker stuff 🐳

**Comment by VKong6019** [comment]
p a i n

**Comment by VKong6019** [comment]
at least i have a different error now :')
`connect ECONNREFUSED 127.0.0.1:5432`

File: README.md [hunk 1]
```diff
@@ -28,12 +28,17 @@ Here is a helpful list of psql commands: https://www.postgresqltutorial.com/psql
...
-To create your config file:
-1. Duplicate the `ormconfig.example.json` file.
-2. Rename the copied file `ormconfig.json`. This will be your config file.
-3. In `ormconfig.json`, fill in values for `"username"` and `"password"` with the credentials of the user your created above. 
-For `"database"`, use the database name you used above.   
+In the pharmd-tracker-backend directory, you should see a file called `ormconfig.ts`. \
+To set up your config file:
+1. Create a `.env` file at the project root directory
+2. Put your secrets in the `.env` file for `POSTGRES_USER`, `POSTGRES_PASSWORD`, `POSTGRES_DB_DEV`, `POSTGRES_DB_TEST`, and `POSTGRES_HOST` like so
+   >POSTGRES_USER=exampleuser\
```
Comment (sophiacrennan) [inline README.md:35]: I didn't need the \ at the end of each line in the .env file

**Review by sophiacrennan — APPROVED** [review]
This is dope. `npm start` and `npm test` still works for me locally and I just created a PR to test that the automated tests run/pass for me, which it does 😎

File: README.md (hunk 1 above)
Comment (VKong6019) [reply README.md:35]: oh the \ is just markdown for a new line, so it shouldn't show up on the actual view

Comment (sophiacrennan) [reply README.md:35]: oooh okok

**Pull request merged**